src/test/java/com/bobocode/hoverla/bring/context/BeanDefinitionValidatorTest.java

```diff
@@ -170,70 +170,73 @@ void checkingSelfDependency() {
     }
 
     @Test
```

Comment:
I rewrote the test below because the previous version was a duplicate of another case. And now it tests the new circular case.